src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/WcfUserNameService.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IdentityModel.Selectors;
+using System.ServiceModel;
+
+namespace WcfService
+{
+    public class CustomUserNameValidator : UserNamePasswordValidator
+    {
+        // This method validates users. It allows in two users, test1 and test2 
+        // with passwords 1tset and 2tset respectively.
+        public override void Validate(string userName, string password)
+        {
+            if (null == userName || null == password)
+            {
+                throw new ArgumentNullException();
+            }
+
+            if (!(userName == "test1" && password == "1tset") && !(userName == "test2" && password == "2tset"))
+            {
+                throw new FaultException("Unknown Username or Incorrect Password");
+            }
```

Comment:
I see the [example] (https://msdn.microsoft.com/en-us/library/system.identitymodel.selectors.usernamepasswordvalidator(v=vs.110).aspx ) this was taken from throws SecurityTokenException. That type is not available in CoreCLR or NET Native. However, it *IS* available in full .NET which is where the service is running. I'd prefer to see the service throw exactly what the example throws to understand how this behaves when it is caught by the client. It would replicate what customers would have done.
And we need to ensure that whatever the client gets back from the call matches what it gets on the desktop.